src/main.c

```diff
@@ -38,920 +38,855 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+        } else {
+            twidth = cwidth * opt.thumb / 100;
+            theight = cheight * opt.thumb / 100;
+        }
+
+        imlib_context_set_anti_alias(1);
+        thumbnail = imlib_create_cropped_scaled_image(0, 0, cwidth, cheight,
+            twidth, theight);
+        if (!thumbnail)
+            err(EXIT_FAILURE, "unable to create thumbnail");
+        else {
+            if (opt.note)
+                scrotNoteDraw(image);
+
+            scrotHaveFileExtension(opt.thumbFile, &haveExtension);
+
+            if (!haveExtension) {
+                imlib_context_set_image(thumbnail);
+                imlib_image_set_format("png");
+            }
 
-      apply_filter_if_required();
+            filenameThumb = imPrintf(opt.thumbFile, tm, NULL, NULL, thumbnail);
+            scrotCheckIfOverwriteFile(&filenameThumb);
 
-      imlib_save_image_with_error_return(filename_thumb, &im_err);
-      if (im_err) {
-        fprintf(stderr, "Saving thumbnail %s failed: %s\n", filename_thumb, strerror(errno));
-	exit(EXIT_FAILURE);
-      }
+            applyFilterIfRequired();
+
+            imlib_save_image_with_error_return(filenameThumb, &imErr);
+            if (imErr) {
+                fprintf(stderr, "Saving thumbnail %s failed: %s\n", filenameThumb, strerror(errno));
+                exit(EXIT_FAILURE);
+            }
+        }
     }
-  }
-  if (opt.exec)
-    scrot_exec_app(image, tm, filename_im, filename_thumb);
-  imlib_free_image_and_decache();
+    if (opt.exec)
+        scrotExecApp(image, tm, filenameIM, filenameThumb);
+    imlib_free_image_and_decache();
 
-  return 0;
+    return 0;
 }
 
-void
-scrot_do_delay(void)
+void scrotDoDelay(void)
 {
-  if (opt.delay) {
-    if (opt.countdown) {
-      int i;
-
-      printf("Taking shot in %d.. ", opt.delay);
-      fflush(stdout);
-      sleep(1);
-      for (i = opt.delay - 1; i > 0; i--) {
-        printf("%d.. ", i);
-        fflush(stdout);
-        sleep(1);
-      }
-      printf("0.\n");
-      fflush(stdout);
-    } else
-      sleep(opt.delay);
-  }
+    if (opt.delay) {
+        if (opt.countdown) {
+            int i;
+
+            printf("Taking shot in %d.. ", opt.delay);
+            fflush(stdout);
+            sleep(1);
+            for (i = opt.delay - 1; i > 0; i--) {
+                printf("%d.. ", i);
+                fflush(stdout);
+                sleep(1);
+            }
+            printf("0.\n");
+            fflush(stdout);
+        } else
+            sleep(opt.delay);
+    }
 }
 
-size_t scrot_have_file_extension(char const *filename, char **ext)
+size_t scrotHaveFileExtension(char const* filename, char** ext)
 {
     *ext = strrchr(filename, '.');
 
-    if (*ext) {
+    if (*ext)
         return strlen(*ext);
-    }
 
     return 0;
 }
 
-
-void scrot_check_if_overwrite_file(char **filename)
+void scrotCheckIfOverwriteFile(char** filename)
 {
-  char *curfile = *filename;
+    char* curFile = *filename;
+
+    if (opt.overwrite == 1)
```

Comment:
`if (opt.overwrite)` , is enough

Reply:
fixed in 04c1b4969d06103ce9aea93ddf7413c75ab5766a